### DISCUSSION TITLE
How can we speed up our pull request workflows? (Goals stage)

### DISCUSSION THREAD

**Discussion by teor2345** [ZcashFoundation/zebra#3935] [Q&A]
Let's have a structured discussion about speeding up pull requests.

(I think this will help us make decisions, because it seemed like we might have some different goals.)

### Introduction

In a structured discussion:
1. we agree on our goals,
2. we agree on some strategies - the ways we can achieve those goals, then
3. we create specific tasks to implement the strategies.

For example:
<details>

If our goal is to have a working Zcash node:
- a strategy is "code review"
- an implementation is "require reviews before merging code"

</details>

This is the first stage, where we focus on *goals*.

### Goals

- We want our code to enforce the consensus rules, and interface with `lightwalletd`
- We want to start new work based on the latest code
- We don't want merge conflicts
- If a PR makes CI fail, we don't want to merge it to `main`, because then everyone's PRs fail CI

### Questions

Are there any goals missing?
Which goals are more important?

**Comment by teor2345** [comment]
I just thought of another goal:
- PRs & CI should work smoothly, so we can focus on code & reviews

**Reply by conradoplg** [reply]
Part of this, maybe: we want chain of PRs to not take too long to merge so that work that depends on them goes smoothly, and to not discourage creating small PRs in the first place

**Comment by oxarbitrage** [comment]
FYI: This PR has been in the merge queue for more than 6 hours and counting: https://github.com/ZcashFoundation/zebra/pull/3908

**Reply by teor2345** [reply]
That was a CI workflow bug, we fixed it in PR https://github.com/ZcashFoundation/zebra/pull/3970

**Comment by dconnolly** [comment]
How are we feeling about this?
